api_types.ts

```diff
@@ -941,13 +950,21 @@ export interface ExecutionContext {
    */
   readonly sync?: Sync;
 
+  /**
+   * If this function is being invoked with authentication, this indicates which authentication was used.
+   *
+   * TODO(patrick): Unhide this
+   * @hidden
+   */
+  readonly authenticationKey?: string;
```

Comment:
Probably need a new name here, "authentication key" is very suggestive of a secret credential.

Reply:
Good point... brainstorming:
- AuthenticationMode
- AuthenticationFlavor
- AuthenticationStrategy
- AuthenticationMethod

And if we don't have the "key" suffix here, maybe we switch the "key" property of AuxiliaryAuth to being "id"?

cc @ekoleda-codaio

Reply:
New idea from @alan-codaio - change "key" to "name" on AuxAuth interface, and then this becomes authenticationName

Reply:
I think `authenticationName` is pretty good. There are other instances where we use `name` to refer to the internal label for a thing:

- `SetEndpoint.name`
- `CustomAuthParameter.name`